### ISSUE TITLE
False positive for explicit_type_interface on switch-case-let

### ISSUE THREAD

**Issue by J-Mendes** [realm/SwiftLint#3615]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

This start to happen after I updated to Xcode 12.5. Basically, all the `switch` statements that have a `let` inside the `case` clause are triggering this rule.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
```

`Explicit Type Interface Violation: Properties should have a type interface (explicit_type_interface)`

### Environment

* SwiftLint version 0.43.1
* Installation method used: Homebrew
* Paste your configuration file:

```yml
opt_in_rules:
  - anyobject_protocol
  - closure_end_indentation
  - closure_spacing
  - explicit_init
  - explicit_self
  - implicit_return
  - joined_default_parameter
  - literal_expression_end_indentation
  - modifier_order
  - number_separator
  - operator_usage_whitespace
  - optional_enum_case_matching
  - prefer_self_type_over_type_of_self
  - prefer_zero_over_explicit_init
  - redundant_nil_coalescing
  - strong_iboutlet
  - test_case_accessibility
  - toggle_bool
  - unneeded_parentheses_in_closure_argument
  - unused_import
  - vertical_whitespace_closing_braces
  - vertical_whitespace_opening_braces

implicit_return:
  included: [closure]
vertical_whitespace:
  max_empty_lines: 2
```

* Which Xcode version are you using? Xcode 12.5 Build version 12E262
* Do you have a sample that shows the issue?

```swift
switch result {
        // This triggers a violation
        case .success(let response):
                ...
        // This triggers a violation
        case .failure(let error):
                ...
}
```

**Comment by marcelofabri** [comment]
I couldn't repro this on Xcode 12.4, so maybe something changed in the SourceKit response in Swift 5.4. I'm downloading Xcode 12.5 and will check later.

**Comment by marcelofabri** [comment]
OK, I was able to repro this with Swift 5.4 and it's indeed because the SourceKit structure has changed.

**Comment by BISARMA** [comment]
Is this fix included in the next release?

**Comment by ppamorim** [comment]
@marcelofabri I am getting the same issue with issue 5.4. :(

**Comment by BrunoCerberus** [comment]
i'm having the same problem T.T

**Comment by daniellangh** [comment]
Same issue here, is there an ETA for the next release? Thanks.
